feat: add spellcheck ci for the docs in the repo

tunakasif opened the pull request:
# Description

A configurable [Check Spelling GitHub Action](https://github.com/marketplace/actions/check-spelling) is provided in the marketplace with MIT License. The implementation is imported and some specific words are whitelisted for the [lunarvim.org](https://github.com/LunarVim/lunarvim.org) repository, e.g., `lvim`. Currently, spellchecking is applied only to "docs" files like `.md` files, but it can be extended to additional files, in the future. Adjustments such as whitelisting certain words, ignoring some patterns can be achieved according to the [provided instructions](https://github.com/check-spelling/check-spelling/wiki).

### Things to Know

- Currently, action is triggered on every `push` and `pull_request`, you may want to restrict this trigger to specific events maybe.
- The base template is copied from [spell-check-this](https://github.com/check-spelling/spell-check-this), and a few LunarVim specific adjustments are made.

### Dependencies:

-  [Check Spelling GitHub Action](https://github.com/marketplace/actions/check-spelling)

fixes #226 

## How Has This Been Tested?

Since this change introduces a GitHub Action, action itself is tested by creating a commit with a known typo and verifying it fails, then fixing the typo and verifying it passes.

kylo252 commented on .github/workflows/spelling.yml:
```diff
@@ -0,0 +1,66 @@
+        task: ${{ needs.spelling.outputs.followup }}
+        experimental_apply_changes_via_bot: 1
+
+  update:
+    name: Update PR
+    permissions:
+      contents: write
+      pull-requests: write
+    runs-on: ubuntu-latest
+    if: ${{
+        github.event_name == 'issue_comment' &&
+        github.event.issue.pull_request &&
+        contains(github.event.comment.body, '@check-spelling-bot apply')
+      }}
+    concurrency:
+      group: spelling-update-${{ github.event.issue.number }}
+      cancel-in-progress: false
+    steps:
+    - name: apply spelling updates
+      uses: check-spelling/check-spelling@main
+      with:
+        experimental_apply_changes_via_bot: 1
+        checkout: true
```
I don't think this will work, since no `ssh_key` is configured AFAIK, maybe use this as a template instead: https://github.com/microsoft/PowerToys/blob/main/.github/workflows/spelling2.yml

tunakasif commented:
@kylo252 I updated workflow according to the [template](https://github.com/microsoft/PowerToys/blob/a397e72e/.github/workflows/spelling2.yml) you provided, and to the [instructions for cspell dictionaries](https://github.com/check-spelling/check-spelling/wiki/Breaking-change:-Smaller-cspell-dictionaries). 

Note that, first I tried to pull the new changes, but the docs action was failing, so I reverted back, that's explanation of that shameful force-push 😅
